### PR TITLE
Allow wait for replications and update replica_regions requirements

### PR DESCRIPTION
**Pull request by yec-akamai**
## 📝 Description

Add a new attribute `wait_for_replications` allow user to wait for image replication to be ready before returning. Default to be false because it's a pretty time consuming process. 

Require at least one available region when updating the `replica_regions` field. 

Add more test cases for image replication.

## ✔️ How to Test

```
make PKG_NAME=linode/image int-test
```

Manual Test:
1. In a sandbox environment, run the following config to perform a image replication
```
resource "linode_instance" "foobar" {
    label = "test-tf-instance"
    type = "g6-standard-1"
    region = "us-east"
    image = "linode/alpine3.19"
    booted = true
}

resource "linode_image" "foobar" {
    linode_id = "${linode_instance.foobar.id}"
    disk_id = "${linode_instance.foobar.disk.0.id}"
    label = "test-gen2-image"
    description = "descriptive text"
    tags = ["test"]
    replica_regions = ["us-east", "us-central"]
    wait_for_replications = true
}
```
2. Observe that image is successfully replicated in 2 regions.
3. Change the replica_regions to use a new region
```
resource "linode_image" "foobar" {
    linode_id = "${linode_instance.foobar.id}"
    disk_id = "${linode_instance.foobar.disk.0.id}"
    label = "test-gen2-image"
    description = "descriptive text"
    tags = ["test"]
    replica_regions = ["eu-west"]
    wait_for_replications = true
}
```
4. Expect to see an error.
5. Change around the config to find any edge case.

### CODE REVIEW

**Review by ezilber-akamai — DISMISSED** [review]
Integration tests are passing, but I'm consistently getting the following error when running step  1 of the manual test:
```
│ Error: Failed to replicate image
│ 
│   with linode_image.foobar,
│   on main.tf line 26, in resource "linode_image" "foobar":
│   26: resource "linode_image" "foobar" {
│ 
│ [400] Image is not ready for replication yet.

```

**Comment by yec-akamai** [comment]
> Integration tests are passing, but I'm consistently getting the following error when running step 1 of the manual test:
> 
> ```
> │ Error: Failed to replicate image
> │ 
> │   with linode_image.foobar,
> │   on main.tf line 26, in resource "linode_image" "foobar":
> │   26: resource "linode_image" "foobar" {
> │ 
> │ [400] Image is not ready for replication yet.
> ```

Good catch! I added wait for image to be ready for the create from Linode images. However, I tested locally and the image was stuck at pending a long time. It could be a temporary backend issue. Will keep testing it and push the fix.

**Review by ezilber-akamai — APPROVED** [review]
Integration tests are passing and manual testing is working as intended. Looks great!

File: linode/image/framework_schema_resource.go [hunk 1]
```diff
@@ -102,7 +102,6 @@ var frameworkResourceSchema = schema.Schema{
 		},
 		"file_hash": schema.StringAttribute{
 			Description: "The MD5 hash of the image file.",
-			Computed:    true,
```
Comment (yec-akamai) [inline linode/image/framework_schema_resource.go:105]: `file_hash` is not a computed attribute.

**Review by lgarber-akamai — APPROVED** [review]
Integration & manual test steps are working well, and the validation seems to be pretty solid. Nice work!
